include/deal.II/base/table.h

```diff
@@ -1903,7 +1903,7 @@ TableBase<N,T>::reinit (const TableIndices<N> &new_sizes,
   // if new size is nonzero: if necessary allocate additional memory, and if
   // not fast resize, zero out all values)
   values.resize_fast (new_size);
-  if (!fast)
+  if (!omit_zeroing_entries)
```

Comment:
For the table it is a bit more complicated. On trivial types, it does indeed call `std::memset` with zero on the memory. On non-trivial classes, it does call the default constructor `T()` (in that case, the default constructor is invoked irrespective of `omit_zeroing_entries` due to the C++ rule book).